botorch/acquisition/knowledge_gradient.py

```diff
@@ -150,12 +150,15 @@ def __init__(
                     "If using a multi-output model without an objective, "
                     "posterior_transform must scalarize the output."
                 )
-        self.sampler = sampler
+        self.sampler: MCSampler = sampler
         self.objective = objective
         self.posterior_transform = posterior_transform
         self.set_X_pending(X_pending)
+        # TODO: Set self.X_pending directly in the __init__ to make this look more
```

Comment:
there is some handling of error emission etc in the `set_X_pending` method - ideally this wouldn't have to be repeated in each `__init__` method